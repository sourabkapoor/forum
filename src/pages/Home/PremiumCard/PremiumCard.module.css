.premium-card {
    margin-bottom: 8px;
}
.premium-card__content {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.premium-card__icon {
  object-fit: contain;
}
.premium-card__button {
  background-color: #ff4500;
  border: none;
  color: var(--primary-color);
  font-family: inherit;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: unset;
  line-height: 17px;
  text-transform: unset;
  min-height: 32px;
  min-width: 32px;
  padding: 4px 16px;
  border-radius: 9999px;
  box-sizing: border-box;
  text-align: center;
  width: 100%;
}

.premium-card__button:hover {
    background-color: #ff2000;
    cursor: pointer;
}

.premium-card__text {
}

.premium-card__title {
  font-size: 12px;
  font-weight: 700;
  line-height: 12px;
  margin-top: 2px;
  margin-bottom: 4px;
  color: var(--secondary-color);
}

.premium-card__description {
  font-size: 12px;
  font-weight: 400;
  line-height: 16px;
}
