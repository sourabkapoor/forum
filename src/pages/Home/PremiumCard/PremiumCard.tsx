import CSSModules from "react-css-modules";
import Card from "../../../components/Card/Card";
import styles from "./PremiumCard.module.css";
import redditShield from "../../../assets/reddit-shield.png";

const PremiumCard: React.FC = () => {
  return (
    <Card>
      <div styleName="premium-card">
        <div styleName="premium-card__content">
          <img
            styleName="premium-card__icon"
            src={redditShield}
            alt="reddit premium shield icon"
          />
          <div styleName="premium-card__text">
            <h3 styleName="premium-card__title">Reddit Premium</h3>
            <p styleName="premium-card__description">
              The best Reddit experience, with monthly Coins
            </p>
          </div>
        </div>
      </div>
      <div styleName="premium-card__button-container">
        <button styleName="premium-card__button">Try Now</button>
      </div>
    </Card>
  );
};

export default CSSModules(PremiumCard, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "ignore",
});
