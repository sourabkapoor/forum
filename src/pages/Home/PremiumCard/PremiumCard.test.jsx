import { queryByPlaceholderText, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import each from "jest-each";
import PremiumCard from "./PremiumCard";

describe("PremiumCard", () => {
  test("snapshot", () => {
    const { asFragment } = render(<PremiumCard />);

    expect(asFragment()).toMatchSnapshot();
  });
});
