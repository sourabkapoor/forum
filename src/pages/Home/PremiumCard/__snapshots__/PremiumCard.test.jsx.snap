// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PremiumCard snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="premium-card"
    >
      <div
        class="premium-card__content"
      >
        <img
          alt="reddit premium shield icon"
          class="premium-card__icon"
          src="reddit-shield.png"
        />
        <div
          class="premium-card__text"
        >
          <h3
            class="premium-card__title"
          >
            Reddit Premium
          </h3>
          <p
            class="premium-card__description"
          >
            The best Reddit experience, with monthly Coins
          </p>
        </div>
      </div>
    </div>
    <div
      class="premium-card__button-container"
    >
      <button
        class="premium-card__button"
      >
        Try Now
      </button>
    </div>
  </div>
</DocumentFragment>
`;
